product_margin_classification/models/product_margin_classification.py

```diff
@@ -0,0 +1,188 @@
+        " cost, as defined: Sale Price = (Cost * (100 + Markup)) / 100\n"
+        "It is computed with the following formula"
+        " Markup = 100 * (Sale Price - Cost) / Cost",
+    )
+
+    profit_margin = fields.Float(
+        string="Profit Margin",
+        compute="_compute_profit_margin",
+        inverse="_inverse_profit_margin",
+        digits=dp.get_precision("Margin Rate"),
+        help="Also called 'Net margin' or 'Net Profit Ratio'.\n"
+        "It is computed with the following formula"
+        " Profit Margin = 100 * (Sale Price - Cost) / Sale Price",
+    )
+
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Company",
+        default=lambda s: s._default_company_id(),
+    )
+
+    template_ids = fields.One2many(
+        string="Products",
+        comodel_name="product.template",
+        inverse_name="margin_classification_id",
+    )
+
+    template_qty = fields.Integer(
+        string="Products Quantity", compute="_compute_template_qty", store=True
+    )
+
+    template_correct_price_qty = fields.Integer(
+        string="Total Products With Correct Price",
+        store=True,
+        compute="_compute_template_different_price_qty",
+    )
+
+    template_incorrect_price_qty = fields.Integer(
+        string="Total Products With Incorrect Price",
+        store=True,
+        compute="_compute_template_different_price_qty",
+    )
+
+    template_too_cheap_qty = fields.Integer(
+        string="Total Products Too Cheap",
+        store=True,
+        compute="_compute_template_different_price_qty",
+    )
+
+    template_too_expensive_qty = fields.Integer(
+        string="Total Products Too Expensive",
+        store=True,
+        compute="_compute_template_different_price_qty",
+    )
+
+    price_round = fields.Float(
+        string="Price Rounding",
+        digits=dp.get_precision("Product Price"),
+        default=lambda s: s._default_price_round(),
+        help="Sets the price so that it is a multiple of this value.\n"
+        "Rounding is applied after the margin and before the surcharge.\n"
+        "To have prices that end in 9.99, set rounding 1, surcharge -0.01",
+    )
+
+    price_surcharge = fields.Float(
+        string="Price Surcharge",
+        digits=dp.get_precision("Product Price"),
+        help="Specify the fixed amount to add or substract(if negative) to"
+        " the amount calculated with the discount.",
+    )
+
+    # Default Section
+    @api.model
+    def _default_company_id(self):
+        return self.env.user.company_id.id
+
+    @api.model
+    def _default_price_round(self):
+        decimal_obj = self.env["decimal.precision"]
+        return 10 ** (-decimal_obj.precision_get("Product Price"))
+
+    # constrains Section
+    @api.constrains("markup")
+    def _check_markup(self):
+        for classification in self:
+            if classification.markup == -100:
```

Comment:
You need to use float_compare as using digit precision

Reply:
Done.